### ISSUE TITLE
Annotated tags are missing attributes

### ISSUE THREAD

**Issue by bmalkowski** [crowdin/mobile-sdk-android#195]
**Describe the bug**

We use Android's [annotations to style](https://developer.android.com/guide/topics/resources/string-resource#example---adding-a-custom-typeface) to style a few of our strings. 

Specifically, we're using the annotations to denote urls that we later underline in our code. Our strings.xml file has entries such as:

```
<string name="login_legal_links"><annotation url="https://example.com">View Terms of Use</annotation> and <annotation url="https://example.com">Privacy Policy</annotation></string>
```

When we the Crowdin SDK to download the translations at run-time, the `annotation` tag is missing all of its properties. The SDK is changing the string into this after its downloaded - note the missing `url="https://example.com"` properties:

```
<string name="login_legal_links"><annotation>View Terms of Use</annotation> and <annotation>Privacy Policy</annotation></string>
```

I verified that when I download the translated files from the Crowdin web portal or CLI, the annotation _are_ correct with the `url` property. It's specifically the SDK that is not working.

I got as far as seeing that it's like the parser that assumes any inner tag in a `<string>` has no properties - https://github.com/crowdin/mobile-sdk-android/blob/master/crowdin/src/main/java/com/crowdin/platform/data/parser/StringResourceParser.kt#L74

**Expected behavior**
I would expect the SDK to fully populate the inner tags' properties.

**Additional context**
We're using v1.5.7 of the SDK.

**Comment by andrii-bodnar** [comment]
Hi @bmalkowski, thanks a lot for the detailed description! I think you're right, the issue is likely caused by the incorrect assumption in the `StringResourceParser.kt`.

By the way, we are happy to [accept contributions](https://github.com/crowdin/mobile-sdk-android/blob/master/CONTRIBUTING.md#contributing) in case you'd like to contribute 🙂

**Comment by bmalkowski** [comment]
I started to dive into this. Adjusting the parser was straightforward and I got the `annotation` tags to parse correctly, but it didn't solve my issue. The spans were missing when I went to look for them yet.

I dug deeper and it looks like the next problem is the [CrowdinResources.getText method](https://github.com/crowdin/mobile-sdk-android/blob/master/crowdin/src/main/java/com/crowdin/platform/CrowdinResources.kt#L74). The `string` var correctly has the annotation tags but the call to `string?.fromHtml()` strips out those tags. From what I can find online, the `Html.fromHtml` method doesn't support the `annotation` tag. Android apparently has this [hidden StringBlock class](https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/content/res/StringBlock.java#347) that handles these tags (and others) when calling `resources.getText` normally to create the spans.

If I have time, I'll try to dig in further and see if I can fix it though I don't have a good idea on _how_ to handle this particular new issue yet. 

I may end up working around it on my end and avoid the `annotation` tag entirely.
